### ISSUE TITLE
Investigate power (CW) appearing on rx only ports

### ISSUE THREAD

**Issue by ramdor** [ramdor/Thetis#228]
This may or may not be related. In addition to a receive only antenna (Beverage) that I attach to
port RX2, I also attach a preamplified and multi coupled Shared Apex Loop on Ant Port 3, and my
 primary transmit output is on Ant Port 1. (I leave Port 2 disconnected as an isolation boost). 
I discovered when using CWX in break in or semi breakin mode, my transmit power appears on
 Ant Port 3, for a split second as the internal relays are switching. Consequently, this power
 burst on what is designated as a "do not transmit" Antenna port, has destroyed a 
strisberg active multicoupler. This all came together for me, while testing the Pure Signal
 bug with the rest of the gang. I ultimately determined that RF was appearing on the 
undesired Ant Port by watching it with a dedicated watt meter. This only happens in 
CW mode as the internal TRX clack back and forth with semi break in delays of approx 250 ms. 

I created a video of the phenomena and will post if none of this makes sense.

_Originally posted by @nj2us in https://github.com/ramdor/Thetis/issues/226#issuecomment-1777149557_

**Comment by nj2us** [comment]
https://www.youtube.com/watch?v=mxWFznqrAvI

A you tube video that demonstrates the anomoly.

Jeff NJ2US

**Comment by ramdor** [comment]
Investigation

1) CWX form uses NetworkIO.SetCWX() to turn CW on/off on the TRX. The cwx form code uses threading so this call is happening from various threads. No configuration of alex/antennas are made.

2) A thread called PollPTT in console.cs is running with a 1 ms delay loop  looking for a number of ptt states, one of which is PTT from the radio which is set when NetworkIO.SetCWX is used in 1) . This sets the front end MOX state only if QSKEnabled is not set.

3) On mox change in console.cs, HdwMOXChanged() is eventually called with a call to Alex.UpdateAlexAntSelection() updating any necessary antenna selections with NetworkIO.SetAntBits()

Issues:

1) There is potential for a delay between 1) + 2) .  Consequently 3) will be delayed sometime 1) is already outputting power

2) QSK will never update the mox state, so no calls to Alex will be made to setup SetAntBits.

**Comment by nj2us** [comment]
Here is a more detailed video (as requested by Richie) showing status of RF on all ports:

https://www.youtube.com/watch?v=o6MuSHUVJ8A

**Comment by nj2us** [comment]
Ramdor asked if I had a clue as to how long this may have been an issue, and I answered thusly,

Richie, it's been an issue for several years, as I have POPPED 4 or 5 of those Strisberg active multi coupler on Port 2 and Port 3, and never understood why until this Pure Signal anomaly showed up, causing me to investigate further, that maybe I was blowing these devices up through the Output port and not the input port on the Multicoupler! I suspect, very few users attach antennas with electronic components on ports 2 or 3, and if they have,  maybe they didnt know that couplers and preamps are blowing because RF is showing up where they least suspect. If you figure that most users may be attaching passive antennas that would igore or radiate the unwanted RF, the OP would probably never realize it.

Jeff NJ2US

**Comment by ramdor** [comment]
Yeah ramdor/me same ;)   It does do it here, but only when using CWX. I do not have a key so cant easily test that, but yes it does put power out to an antenna port that is specified as RX only. This may be a firmware thing, not entirely sure. Will have to get the others involved who know more about these things. A warning should be put out tbh.

**Comment by Roturbo** [comment]
Hello, 

i have a Anan 100D **and maybe is not related**, but on some occasions i notice one relay activated
or released inside the Anan most during TX operation and less on RX.

Most of the time i use SSB and notice this, but once on FM i heard very clear on a audio
pause and i make a new pause trying to see if anything have change on power swr etc.


Since this doesn´t present any changes on TX i didn´t care and i have upload the firmware
again to see if it goes way but from time to time some relay is activated, reading this topic
i remember to post it just in case other have same situation and also want to report. 


Regards

**Comment by ramdor** [comment]
I have decided to undo the changes I made Jeff, until we formulate a better fix. The issue is caused because SetCWX used by the cwx form, and serial pin keying causes the radio to output power before the hardware antenna ports etc have been configured. 

The hardware is configured after Thetis detects a cw hardware ptt change, and this obviously will be after power is coming out because of the SetCWX.

A new branch will be created at some point to work on this alone.

**Comment by ramdor** [comment]
didnt want to close this, oops.

**Comment by w3ub** [comment]
I ran an RX antenna a couple of years ago, and burnt out the MOSFET in my RX preamplifier about four times, always in CW mode ... I think this was the cause.
Doug

**Comment by RdWing** [comment]
I believe this is the same issue I was seeing with my Mk 2.5 7000. 
See [post](https://community.apache-labs.com/viewtopic.php?f=9&t=4382&p=22287#p22287) on the apache forum from last October 2022.

Note, I am specifically speaking about power being detected (and actually transmitted), on other antenna ports.

I don't think I was using serial port keying at that time.

**Comment by ramdor** [comment]
@RdWing yes, this is that is exactly this same issue.  If you watch Jeff NJ2US excellent vids further up this thread they show the same issue.

**Comment by n1gp** [comment]
I have verified this also happens on PiHPSDR.
It happens on an ANAN-G2.
It happens using an Iambic key plugged in to the front,
and also with a straight key.

Seems to be on the transition from RX->TX

Settings:

**Comment by dl1ycf** [comment]
I think I understand what's happening.

Here I refer to the Anan-7000 PA board but the situation
is the same for other boards.

Suppose Ant3 is used for RX and Ant1 for TX.

Upon a RX/TX transition, two relays are switched:

K3 which is closed during RX opens  (disconnects Ant3 from T/R circuit)
K1 which is open   during RX closes (connects    Ant1 to   T/R circuit)

Relay switching times are relatively slow, of the order 20 msec

If no RF appears quite fast after switching time (and since the T/R
circuit in Anan-7000 is with diodes it is likely) then some msec
of RF  output may still find their way to Ant3 since K3 is not 
completely opened.

This is in particular a problem with "CW handled in Radio" for the following
reason:

- you touch the CW paddle
- the FPGA puts the radio into TX mode and sets the "PTT" flag
- PTT flag is sent to host PC and CW transmission starts
- host PC slews down receivers and slews up TX,
  and then it changes the relays.

I do not see how to solve the problem from the software side in this  case,
since the RX/TX switching as well as  the RF pulse  formation is 
entirely done in the FPGA. So the SDR program is informed about the
T/R switch while it is already going on and therefore the commands
to change the relays comes late (even if you do it in the SDR program
immediately after receiving the "PTT on" in the HPSDR data stream.

If the RX/TX switch is initiated in the SDR program, one can certainly
include a delay between antenna switching and setting PTT, but this
case is  not so dangerous anyway since the TX IQ samples have to make
their way through the buffers. The "CW handled in Radio" case
it where I have to scratch my head.

The "CWX" case in Thetis should be manageable, since the SDR program sends the
"PTT on", doesn't it?

DL1YCF

**Comment by dl1ycf** [comment]
In my preferred CW setup the problem is much less likely to appear:

- use and external (WinKey) keyer
- connect Key *and* PTT of  the keyer with the CW and PTT in jacks
- use a "PTT lead in" delay of 150 msec

This way, the radio is put into TX mode 150msec before generating RF, and this is usually enough to inform
the SDR program that we are going TX and the SDR program then has time  to switch the Ant1/Ant3 relays.

**Comment by laurencebarker** [comment]
Hi Rick

 

QSK can’t operate with different antennas. I would have expected it to refuse to enter QSK state with split antennas. Thetis never gets to know it is in TX, so never changes the antennas. So the condition you found is definitely bad!

 

I suspect a full solution will need protocol 2 message change to tell the FPGA the required antenna for TX and RX, so it can initiate instant changeover during the delay period. But I doubt we want to go that far!

 

In semi breakin antenna changeover will be delayed because it requires a message exchange to Thetis and from Thetis before the relays will be driven to the TX state. Increasing the CW delay may help (Thetis sets 7ms before the CW ramp starts, and the antenna change will definitely still be in progress).

 

 

 

 

Laurence Barker G8NJJ

***@***.***

**Comment by ramdor** [comment]
I have changed this, to ignore waterfall pixels during the fft fill time. Blank single pixel rows are added instead. I will make a video in due course to highlight things I have discovered. I can also confirm that on my 7000dle2 there is a pulse of rf which can be seen on an external SDR and is now shown in this waterfall change IF rf delay is enough to push it beyond the fft fill time.

**Comment by w-u-2-o** [comment]
Doesn't this go against #204?

**Comment by nj2us** [comment]
My band aid approach to masking this problem is to attach my "Active" receive antenna to the "EXT" port, and then set Receive Antenna Control to EXT1. I couldnt find any operational mode that cause Transmit RF to escape through EXT1. The only functionality lost is that you can't do a fast toggle between TX and RX antennas on the VFO control/Band Stack window in the GUI. Maybe the EXT1 port can be manipulated in software to allow quick toggle between EXT1 and Transmit Antennas.

**Comment by n4py** [comment]
I would think setting the CW key down value to 15 msec would solve this.

**Comment by n4py** [comment]
So I just got an Anan 8000DLE.  I can confirm this is a serious problem.  I did the following experiment.
1) I connected my LP-500 watt meter to antenna 3.
2) I put a dummy load on antenna 1.
3) I set antenna 1 for xmit and antenna 3 for receive.
4) I transmitted 100 watts with my paddles via an external keyer.

Many times I saw a brief 100 watts on antenna 3.  Also saw a brief high SWR several times.  This also failed in the same way sending CW via an external software program sending KY commands.

So there is definitely a bad timing problem here.  I had CW key down delay set at 15 msec.  I was using semi breaking at 250 msec delay.

Carl N4PY

**Comment by w-u-2-o** [comment]
It doesn't appear to be a hardware problem, and if that's indeed the case
then Apache is not to blame.

All relays are controlled by software and firmware. When they switch or
don't switch is up to the code.

**Comment by mdblack98** [comment]
By definition a relay is hardware.
However you are correct that control of them is firmware.

However, I would maintain this is the radio's firmware that is allowing power to go out an unselected relay.

I don' think it's Thetis that is telling the rig to turn off the RX antenna while transmitting, is it?  I think that's the rig's responsibility.  Software should never be allowed to do such things as to cause hardware damage.

Also it's a matter of port isolation during transmit.  You should NOT be able to send much power out a 2nd unselected port for any length of time.
It's a matter of relay settling to disable the RX port right now.   So is the power being seen coming from port cross over or the RX antenna seeing the power being transmitted?

I was just watching VE2DX's video and him showing 80dB of port isolation on his antenna switch.

https://www.youtube.com/watch?v=Sav6k4v9mT4

Mike W9MDB



On Wednesday, January 3, 2024 at 08:25:26 AM CST, w-u-2-o ***@***.***> wrote: 






It doesn't appear to be a hardware problem, and if that's indeed the case 
then Apache is not to blame. 

All relays are controlled by software and firmware. When they switch or 
don't switch is up to the code.

**Comment by dl1ycf** [comment]
If it comes to the hardware, ANT 3 is not an "RX only port" (it is in the TX chain).
You have bad luck since for the Anan-8000, only the "PS Input" port is "RX only"
(the "Ext1" and "Xvrt In", present in the Anan-7000, have been sacrificed).

One can however (nearly) solve the problem in the SDR program. I have done so
in piHPSDR, so you can use that program. The Thetis people think it must be
solved in the firmware. While it is not unreasonable to demand this,
it will take time.

**Comment by mdblack98** [comment]
I just checked the code and I see this

 Alex.getAlex().setRxOnlyAnt(band, (byte)ant);

This indicates the RX only selection is controlled by Alex.c

This appears to be controlled in UpdateAlexAntSelection

I see a SendHighPriority which no doubt has some meaning.

Also see SetAntBits but not GetAntBits where one could check status.

Is the handling of this socket request asynchronous in the rig?
Do we need a delay after SetAntBits?

Mike W9MDB

**Comment by laurencebarker** [comment]
Hi Mike

 

Actually yes: it is Thetis telling it to turn off the RX antenna and select the TX antenna.

 

I did write down the sequence hen this was first raised, but following a PC failure I’ve lost my entire email history so I’ll have to do this again.

 

In QSK mode: Thetis isn’t involved. QSK should only be selected if RX antenna = TX antenna

 

With no break-in: Thetis tells the hardware to enter TX mode with TX antenna BEFORE key down, so the issue doesn’t arise. 

 

That leaves semi-break in CW, which is where the issue occurs. The issue happens because protocol 2 has a field to select the antenna in use (ANT1-3); it does not have separate data for RX antenna and TX antenna. Thetis tell the firmware which antenna to use; and it needs to send a message to change over between TX and RX antennas if they are different. 

 

In the FPGA firmware, beginning in an RX state:

*	The user presses the key down
*	The keyer counts a delay before taking any action (Thetis default I think was 7ms)
*	The keyer enters TX mode, and signals Thetis
*	The keyer ramps up the CW power level over around 5ms
*	When a high priority message arrives from Thetis, the firmware selects the TX antenna

 

Thetis receives the message that the key is down, selects its TX state and sends a message a while later to select the correct TX antenna.

 

A “thorough” fix would be to extend the protocol 2 message to send RX antenna and TX antenna, so the firmware can decide immediately which to use. BUT that would require protocol 2 firmware change. Christoph has implemented a software-only solution that enabled piHPSDR to achieve a much more rapid transition, and with a longer “delay before starting to TX” this does seem to solve the problem.

 

 

Laurence Barker G8NJJ

***@***.***

**Comment by mdblack98** [comment]
In my case I know what happened to me.
I don't normally do CW.  So I was happily operating along and then started working on improvements to FLDigi for the frequency measurement test.
During that time I started doing CW with Thetis and I now know that's when it blew out my Pixel Pro preamplifier.
At the time I didn't know that....thought it was just time for the amplifier to go...got it repaired and it blew a 2nd time (still testing CW at my station).
I now know not to do CW with my RX only antenna connected.

This should be a high priority to get fixed in Protocol 2.

Mike W9MDB








On Wednesday, January 3, 2024 at 09:27:00 AM CST, Laurence Barker ***@***.***> wrote: 






Hi Mike 



Actually yes: it is Thetis telling it to turn off the RX antenna and select the TX antenna. 



I did write down the sequence hen this was first raised, but following a PC failure I’ve lost my entire email history so I’ll have to do this again. 



In QSK mode: Thetis isn’t involved. QSK should only be selected if RX antenna = TX antenna 



With no break-in: Thetis tells the hardware to enter TX mode with TX antenna BEFORE key down, so the issue doesn’t arise. 



That leaves semi-break in CW, which is where the issue occurs. The issue happens because protocol 2 has a field to select the antenna in use (ANT1-3); it does not have separate data for RX antenna and TX antenna. Thetis tell the firmware which antenna to use; and it needs to send a message to change over between TX and RX antennas if they are different. 



In the FPGA firmware, beginning in an RX state: 

* The user presses the key down 
* The keyer counts a delay before taking any action (Thetis default I think was 7ms) 
* The keyer enters TX mode, and signals Thetis 
* The keyer ramps up the CW power level over around 5ms 
* When a high priority message arrives from Thetis, the firmware selects the TX antenna 



Thetis receives the message that the key is down, selects its TX state and sends a message a while later to select the correct TX antenna. 



A “thorough” fix would be to extend the protocol 2 message to send RX antenna and TX antenna, so the firmware can decide immediately which to use. BUT that would require protocol 2 firmware change. Christoph has implemented a software-only solution that enabled piHPSDR to achieve a much more rapid transition, and with a longer “delay before starting to TX” this does seem to solve the problem. 





Laurence Barker G8NJJ 

***@***.***

**Comment by dl1ycf** [comment]
Michael, you miss the point.

If CW is handled in radio, and the key is pressed, then the radio
goes TX and tells the SDR program about it. The SDR program 
can then react, but this is possibly too late.

**Comment by dl1ycf** [comment]
> 
> This should be a high priority to get fixed in Protocol 2. 
> 

Exactly, a solution is easy in the SDR program, provided
that the protocol defines what one should do. One must have
*two* "Alex words" in the protocol, one for RX and one for
TX. Then, the TX settings can be sent to the radio in 
advance, and the firmware restores these setting
(when doing CW) *before* going TX.

**Comment by mdblack98** [comment]
OK...so let's fix the firmware....





On Wednesday, January 3, 2024 at 10:13:50 AM CST, dl1ycf ***@***.***> wrote: 






Michael, you miss the point. 

If CW is handled in radio, and the key is pressed, then the radio 
goes TX and tells the SDR program about it. The SDR program 
can then react, but this is possibly too late.

**Comment by n1gp** [comment]
I will take this on for putting a fix in to the firmware. I'll look at the current protocol 2 specifications and try to find the best approach to adding the  necessary updates for this issue. I'll update here as well as send out an email to all here and others I think should have input.

**Comment by laurencebarker** [comment]
I'm already looking at the necessary  G2 firmware change but that has the luxury that a processor can easily move the bits around.There are 80 sets of alex words in the high priority message. Could we use one of those eg the one that has 16 bits used for RX2 filter settings? Or just 3 unused bits somewhere for TX ant1-3?Backward compatibility might be an issue.Laurence BarkerSent from my Galaxy
-------- Original message --------From: Rick Koch ***@***.***> Date: 03/01/2024  21:37  (GMT+00:00) To: ramdor/Thetis ***@***.***> Cc: Laurence Barker ***@***.***>, Comment ***@***.***> Subject: Re: [ramdor/Thetis] Investigate power (CW) appearing on rx only ports
  (Issue #228) 
I will take this on for putting a fix in to the firmware. I'll look at the current protocol 2 specifications and try to find the best approach to adding the  necessary updates for this issue. I'll update here as well as send out an email to all here and others I think should have input.

**Comment by n1gp** [comment]
I was looking at using 3 unused bits that are common in each of the Alex0 words for all ANAN boards.  Give them to the
Tx ANT and use the previous ANT 1-3 for Rx. Updated Doc for reference:

[openHPSDR Ethernet Protocol v3.13.docx](https://github.com/ramdor/Thetis/files/13823913/openHPSDR.Ethernet.Protocol.v3.13.docx)

**Comment by dl1ycf** [comment]
Nope. There is nothing to fix in the firmware.

The firmware is OK if it behaves according to the protocol.
It is the protocol that needs be extended, *that* is the problem.


> Am 03.01.2024 um 17:16 schrieb Michael Black ***@***.***>:
> 
> 
> OK...so let's fix the firmware.... 
> 
> 
> 
> 
>

**Comment by dl1ycf** [comment]
It does not make much sense to (just) extend the Alex word, since in principle each bit here
drives its "own" relay.

What one needs is that the firmware automatically changes TX ANT 1/2/3  when going TX.

If you want to put this into the Alex word, you actually need FOUR "unused bits", one
flagging that the other three are valid (backwards compatibility problem).

I think one rather should look at the HighPrio packet. This has eight 32-bit
Alex words (Alex7 ... Alex0) at position 1404 ... 1435.

Here is ample space e.g. re-naming them

Alex3TX, ..., Alex0TX, Alex3, ..., Alex0


Alex?TX will normally be zero (for currently existing SDR programs), and the
firmware could then do then following:

If (transmitting && Alex0TX != 0) {
  clock out Alex0TX word to Alex0 shift register
} else {
  clock out Alex0 word to Alex0 shift register
}

and the same for Alex1. Up do now, Alex2-7 are unused anyway so "restricting"
the protocol from 8 to 4 ALEX banks is probably OK.

Then (and only then), SDR programs can start to update their "Alex logic"
and populate more bits when sending the HighPrio packet.

Just my $0.02,

Christoph DL1YCF.

**Comment by laurencebarker** [comment]
A better description might be "modify the firmware to implement a revised definition of the protocol" and that is now happening.Laurence BarkerSent from my Galaxy
-------- Original message --------From: dl1ycf ***@***.***> Date: 04/01/2024  08:14  (GMT+00:00) To: ramdor/Thetis ***@***.***> Cc: Laurence Barker ***@***.***>, Comment ***@***.***> Subject: Re: [ramdor/Thetis] Investigate power (CW) appearing on rx only ports
  (Issue #228) 
Nope. There is nothing to fix in the firmware.



The firmware is OK if it behaves according to the protocol.

It is the protocol that needs be extended, *that* is the problem.





> Am 03.01.2024 um 17:16 schrieb Michael Black ***@***.***>:

> 

> 

> OK...so let's fix the firmware.... 

> 

> 

> 

> 

>

**Comment by laurencebarker** [comment]
True but many of the relay output bits are not used. Having a 4th bit to identify this scheme is in use is probably essential though. Laurence BarkerSent from my Galaxy
-------- Original message --------From: dl1ycf ***@***.***> Date: 04/01/2024  08:30  (GMT+00:00) To: ramdor/Thetis ***@***.***> Cc: Laurence Barker ***@***.***>, Comment ***@***.***> Subject: Re: [ramdor/Thetis] Investigate power (CW) appearing on rx only ports
  (Issue #228) 
It does not make much sense to (just) extend the Alex word, since in principle each bit here

drives its "own" relay.



What one needs is that the firmware automatically changes TX ANT 1/2/3  when going TX.



If you want to put this into the Alex word, you actually need FOUR "unused bits", one

flagging that the other three are valid (backwards compatibility problem).



I think one rather should look at the HighPrio packet. This has eight 32-bit

Alex words (Alex7 ... Alex0) at position 1404 ... 1435.



Here is ample space e.g. re-naming them



Alex3TX, ..., Alex0TX, Alex3, ..., Alex0





Alex?TX will normally be zero (for currently existing SDR programs), and the

firmware could then do then following:



If (transmitting && Alex0TX != 0) {

  clock out Alex0TX word to Alex0 shift register

} else {

  clock out Alex0 word to Alex0 shift register

}



and the same for Alex1. Up do now, Alex2-7 are unused anyway so "restricting"

the protocol from 8 to 4 ALEX banks is probably OK.



Then (and only then), SDR programs can start to update their "Alex logic"

and populate more bits when sending the HighPrio packet.



Just my $0.02,



Christoph DL1YCF.

**Comment by dl1ycf** [comment]
Hi Rick

 

Either way would work. And yes detecting that txant1-3 contain 001, 010 or 100 would be sufficient to know that it had come from an updated version of Thetis or piHPSDR. Whichever set of bits is used, we need some kind of detection that it is from a modified client app.

 

My job is easiest as I have a processor available to move the bits around.

 

My Verilog IP is different from Orion’s – partly because it has an AXI4-Lite bus interface to get the input data. What I will implement is a second 16 bit word for the 16 bit shifted settings that include the TX filter and ANT bits. But I can easily get the right data to that register. 

 

 

 

Laurence Barker G8NJJ

***@***.***

**Comment by dl1ycf** [comment]
Hi Laurence,

I think what I will do, regarding the current P2 spec, is take the unused
1428 & 1429 Alex 1 16 bit word
and make that the TxAlex equal to 1434 & 1435 Alex 0 16 bit word but
containing the new TX ANT bits.

1428

Alex 1

Reserved for future use

1429

Alex 1

Reserved for future use

1430

Alex 1

[15:8] Orion Mk 11 (ANAN-8000DLE)

1431

Alex 1

[7:0] Orion Mk 11 (ANAN-8000DLE)

1432

Alex 0

[31:24]

1433

Alex 0

[23:16]

1434

Alex 0

[15:8]

1435

Alex 0

[7:0]

In the ANAN firmware I will just switch to that word if it is valid on
transmit.

Unfortunately with the 'free' version of quartus, as you know, it does not
allow any
logic locking. So every compile is new and quartus will many times make a
different
decision on how to route a signal and/or clocks. So the chances of getting
firmware
that works just the same as before but with this new Alex switching is
going to be
hit or miss.

**Comment by rz3qs** [comment]
Thetis has a very old mistake.
Why this is done, I do not understand.
The speed of the CW and the correct time of switching the relay.
Speed CW is one function, the control time ms is another function.

https://drive.google.com/file/d/1Y9eTaGl02th_bRaOREp3fZD-Mn_2l32f/view?usp=sharing

**Comment by laurencebarker** [comment]
This is not a Thetis problem. The problem is the design of the protocol. There is (and never was) any guarantee about how long it takes ethernet messages to arrive; nevertheless the protocol requires a message exchange to change antenna.

 

Changes to Thetis might get to the point where the problem was “largely resolved”. But only a change to the protocol can resolve it completely.

 

 

Laurence Barker G8NJJ

***@***.***

**Comment by dl1ycf** [comment]
Just an update on this. I want to correct what I last said,
'take the unused 1428 & 1429 Alex 1 16 bit word
and make that the TxAlex equal to 1434 & 1435 Alex 0 16 bit word but
containing the new TX ANT bits.'

I meant,
'take the unused 1428 & 1429 Alex 1 16 bit word
and make that the TxAlex equal to 1432 & 1433 Alex 0 16 bit word but
containing the new TX ANT bits.'

I gather from what you said Laurence, that is how you're implementing this
as well? We'll have to do
the same thing with what's exposed to the client.

I have some modified Orion2 firmware and have made some modifications to
pihpsdr for testing it.
I went back to Nov 2nd 2023 before Christoph added support to mitigate the
RF on RX-only ANT selection.
Then added this:

diff --git a/src/new_protocol.c b/src/new_protocol.c
index 5028f74..84d2e53 100644
--- a/src/new_protocol.c
+++ b/src/new_protocol.c
@@ -1178,6 +1178,27 @@ static void new_protocol_high_priority() {
     break;
   }

+#if 1
+  // use upper 16 bits of alex1 for TX relay control like ANT selection
+  i = transmitter->alex_antenna;
+  alex1 |= alex0 & 0xf8ff0000; // copy upper alex0 except for ANT1-3
+  switch (i) {
+  case 0:  // ANT 1
+    alex1 |= ALEX_TX_ANTENNA_1;
+    break;
+
+  case 1:  // ANT 2
+    alex1 |= ALEX_TX_ANTENNA_2;
+    break;
+
+  case 2:  // ANT 3
+    alex1 |= ALEX_TX_ANTENNA_3;
+    break;
+  }
+
+  high_priority_buffer_to_radio[1428] = (alex1 >> 24) & 0xFF;
+  high_priority_buffer_to_radio[1429] = (alex1 >> 16) & 0xFF;
+#endif
   high_priority_buffer_to_radio[1432] = (alex0 >> 24) & 0xFF;
   high_priority_buffer_to_radio[1433] = (alex0 >> 16) & 0xFF;
   high_priority_buffer_to_radio[1434] = (alex0 >> 8) & 0xFF;

I no longer see any RF when using an RX-only ANT.

Laurence, have you had a look at what it would take to modify Thetis for
this?

**Comment by n1gp** [comment]
The above was from an email I sent, not dl1ycf.

**Comment by dl1ycf** [comment]
Hi Rick

 

Because of an ongoing family issue I’m not getting much time for radio stuff!

 

As far as I can see, the TX antennas are already accessible in alex.cs. I think NetworkIO.SetAntBits() needs a new parameter for TX antenna (at the moment is passes down the ANT1-3 bits already selected for RX or TX). Then it can be saved where the protocol 2 code can access it. Then network.c around line 809 needs to set the 2 new bytes.

 

This will need a new release because the DLL changes, so we will need to synchronise with Richie.

 

I was going to do my FPGA code update 1st which is partly why I hadn’t got far. I imagine it will be late in the coming week before I have FPGA code and updated p2app.

 

 

Laurence Barker G8NJJ

***@***.***

**Comment by n1gp** [comment]
Firmware with the discussed fix is available for the ANAN-7000/8000 here:
https://www.dropbox.com/scl/fi/6rs685y07wlljjuhtj8sd/Orion_MkII_Protocol_2_v2.1.27.rbf?rlkey=3athz5i9t0k84ciy65o1cqqip&dl=0

**Comment by mdblack98** [comment]
Do we know if this is a problem with the G2 and G2-1K?My G2-1K is in the pipeline for estimated delivery in Feb right now.
Mike W9MDB

 

    On Monday, January 8, 2024 at 06:40:38 AM CST, Rick Koch ***@***.***> wrote:  
 
 


Firmware with the discussed fix is available for the ANAN-7000/8000 here:
https://www.dropbox.com/scl/fi/6rs685y07wlljjuhtj8sd/Orion_MkII_Protocol_2_v2.1.27.rbf?rlkey=3athz5i9t0k84ciy65o1cqqip&dl=0

**Comment by n4py** [comment]
Yes, I would think it is certainly also a problem with the G2.

Carl Moreschi N4PY
127 River Moss Way
Hertford, NC 27944
www.n4py.com

**Comment by mdblack98** [comment]
So what firmware does the G2 use?
Mike W9MDB

 

    On Monday, January 8, 2024 at 08:51:33 AM CST, n4py ***@***.***> wrote:  
 
 

Yes, I would think it is certainly also a problem with the G2.

Carl Moreschi N4PY
127 River Moss Way
Hertford, NC 27944
www.n4py.com

**Comment by n1gp** [comment]
I know Laurence G8NJJ is working on a fix for this and the G2 but is tied up with other things
at the moment.
His repo for the firmware is here: https://github.com/laurencebarker/Saturn

**Comment by laurencebarker** [comment]
G2 is totally different firmware from 7000 and 8000. But it will be fixed this month! It is only a small module and small change.Laurence BarkerSent from my Galaxy
-------- Original message --------From: Michael Black ***@***.***> Date: 08/01/2024  14:57  (GMT+00:00) To: ramdor/Thetis ***@***.***> Cc: Laurence Barker ***@***.***>, Comment ***@***.***> Subject: Re: [ramdor/Thetis] Investigate power (CW) appearing on rx only ports
  (Issue #228) 
So what firmware does the G2 use?

Mike W9MDB



 



    On Monday, January 8, 2024 at 08:51:33 AM CST, n4py ***@***.***> wrote:  

 

 



Yes, I would think it is certainly also a problem with the G2.



Carl Moreschi N4PY

127 River Moss Way

Hertford, NC 27944

www.n4py.com

**Comment by w-u-2-o** [comment]
> Firmware with the discussed fix is available for the ANAN-7000/8000 here: https://www.dropbox.com/scl/fi/r8o8iictmcb0qbaka99w7/Orion_MkII_Protocol_2_v2.2.1.rbf?rlkey=8ca3ns4y0u0mrmox8zsl2zb55&dl=0

Rick--it says "file deleted". I'm guessing there was something wrong with it?

**Comment by n1gp** [comment]
I just tried it and its working for me. My first post had the wrong URL make sure you use:
https://www.dropbox.com/scl/fi/r8o8iictmcb0qbaka99w7/Orion_MkII_Protocol_2_v2.2.1.rbf?rlkey=8ca3ns4y0u0mrmox8zsl2zb55&dl=0

and NOT:
https://www.dropbox.com/scl/fi/6rs685y07wlljjuhtj8sd/Orion_MkII_Protocol_2_v2.1.27.rbf?rlkey=3athz5i9t0k84ciy65o1cqqip&dl=0

**Comment by dl1ycf** [comment]
The problem is in principle there for *every* radio that does not
conform to the VERY recently updated protocol standard.


> Am 08.01.2024 um 15:44 schrieb Michael Black ***@***.***>:
> 
> 
> Do we know if this is a problem with the G2 and G2-1K?My G2-1K is in the pipeline for estimated delivery in Feb right now. 
> Mike W9MDB 
> 
> 
>

**Comment by ramdor** [comment]
Quick test here, changes pushed to master from Rick, and a SetCWX fix.

https://youtu.be/XwrvUuHtcEU

notes: 7000dleMK2, paddle/etc not tested, keying via serial not tested

**Comment by ramdor** [comment]
Thetis installer for those wanting to test before dev_1 release is made :

https://www.dropbox.com/scl/fi/woswow2qkcof9i5szvw8t/Thetis-v2.10.3.6-dev_ant_isolation_test.x64.msi?rlkey=sf9q21jvv96ij5nw9zpduouf7&dl=0

note: it is the same version as the one used in the video above, but tagged with 'dev_ant_isolation_test' in the title bar

VT for the above msi : https://www.virustotal.com/gui/file/bf405c98076346f42de7d33514292fff127446dfb62eed3df02c4d5e58f4d495?nocache=1

**Comment by laurencebarker** [comment]
Hi RichieI also want to sneak in a minor change that will affect the DLLs. To pass the CAT over TCPIP server port number in a high priority message. Unlikely to get to this before the weekend because I have G2 firmware for the CW mod to test first. Will that be in timeFor your dev release please?Laurence BarkerSent from my Galaxy
-------- Original message --------From: Richie ***@***.***> Date: 10/01/2024  05:58  (GMT+00:00) To: ramdor/Thetis ***@***.***> Cc: Laurence Barker ***@***.***>, Comment ***@***.***> Subject: Re: [ramdor/Thetis] Investigate power (CW) appearing on rx only ports
  (Issue #228) 
Thetis installer for those wanting to test before dev_1 release is made :
https://www.dropbox.com/scl/fi/woswow2qkcof9i5szvw8t/Thetis-v2.10.3.6-dev_ant_isolation_test.x64.msi?rlkey=sf9q21jvv96ij5nw9zpduouf7&dl=0
note: it is the same version as the one used in the video above, but tagged with 'dev_ant_isolation_test' in the title bar
VT for the above msi : https://www.virustotal.com/gui/file/bf405c98076346f42de7d33514292fff127446dfb62eed3df02c4d5e58f4d495?nocache=1

**Comment by ramdor** [comment]
@laurencebarker yeah mate. No probs. I haven't had much spare time since the new year so haven't really done much, not enough for the dev_1 release to be honest. I just wanted to get something built here so testers could have a go with Rick's new FW and the quick change to SetCWX. So yep, no rush on the next 'true' dev release. Cheers.

**Comment by laurencebarker** [comment]
Perfect thank you. I will get onto it soon!Laurence BarkerSent from my Galaxy
-------- Original message --------From: Richie ***@***.***> Date: 10/01/2024  08:13  (GMT+00:00) To: ramdor/Thetis ***@***.***> Cc: Laurence Barker ***@***.***>, Mention ***@***.***> Subject: Re: [ramdor/Thetis] Investigate power (CW) appearing on rx only ports
  (Issue #228) 
@laurencebarker yeah mate. No probs. I haven't had much spare time since the new year so haven't really done much, not enough for the dev_1 release to be honest. I just wanted to get something built here so testers could have a go with Rick's new FW and the quick change to SetCWX. So yep, no rush on the next 'true' dev release. Cheers.

**Comment by n4py** [comment]
This firmware will not load on my 8000dle

⁣Carl N4PY

Get BlueMail for Android ​

**Comment by n1gp** [comment]
@RdWing 
These two images are for the EP4CE115F29C8N Cyclone IV series FPGA.
I have no way of testing these so please try them both and let me know how it goes.

https://www.dropbox.com/scl/fi/4kt9ipsikj2hpo5mt9560/Orion_MkIII_Protocol_2_v2.2.2A.rbf?rlkey=gnzxr4hw8oj2e07ymzzxnxk9p&dl=0

https://www.dropbox.com/scl/fi/raat7uaypmeqxw8lw5aj6/Orion_MkIII_Protocol_2_v2.2.2B.rbf?rlkey=8tltgz4bho2ohv1y4juzdinlq&dl=0

@n4py 
Does it not complete the programming of the flash or does it complete but not work?
This link may be of some help loading firmware, just in case, I'm not sure how many times
you have performed the upgrades (I suggest using the Bootloader method):
https://community.apache-labs.com/viewtopic.php?f=18&t=2333

**Comment by n4py** [comment]
Neither of these files will load.  They both say they do not match.

Carl Moreschi N4PY
127 River Moss Way
Hertford, NC 27944
www.n4py.com

**Comment by RdWing** [comment]
> Neither of these files will load. They both say they do not match. Carl Moreschi N4PY 

@n4py Those files are for the EP4CE versions of the Orion Mk2 SDR board (Generally found only in 7000 DLE Mk2's) . I believe Rick linked you to the proper thread on doing firmware updates on the 8000.

**Comment by RdWing** [comment]
> @RdWing These two images are for the EP4CE115F29C8N Cyclone IV series FPGA. I have no way of testing these so please try them both and let me know how it goes.

A: refuses to load at all, no IP, etc.
B: Everything works, WB, PS, ADC1, diversity, no seq errors, ant1-3. 
Except, the transmit spectra is quite a bit less pure than it was on the previous 2.1.27G version from the QSK fix era.

New firmware B:![Screenshot 2024-01-12 130804 (Small)](https://github.com/ramdor/Thetis/assets/19214916/e4c8b48b-05db-4289-9b23-3925a24b7a83)
Old Firmware G![Screenshot 2024-01-12 131340 (Small)](https://github.com/ramdor/Thetis/assets/19214916/f68b59f7-19b9-4722-9a9f-a85759a58543)

**Comment by n1gp** [comment]
@RdWing 
Tnx for testing those, I will make a couple more tonight and post them tomorrow.
I'm also going to start a couple of threads on the Apache-Labs forum to track these
images and testing as in this Git-Issues I don't think is the right place.

**Comment by n4py** [comment]
After reading the article about updating FW on my 8000, I saw where the 
name of the file has to start with Metis.  So I renamed the file and it 
loaded fine on my 8000.

I then did the test using FW v2.2.1 and Thetis 2.10.3.6 
dev_ant_isolation_test.

I am happy to report all is working great.  No more short transmit 
bursts on the receive antenna using CW both from my external keyer and 
also using the KY CAT interface with my N4PY software.

Carl Moreschi N4PY
127 River Moss Way
Hertford, NC 27944
www.n4py.com

**Comment by n4py** [comment]
Well after running the new FW and Thetis with my 8000, there is a problem with RX2.  If I set RX2 to 10 meters, it is fine.  But if I then exit Thetis and restart Thetis, RX2 is not receiving any signals.  If I then change RX to 20 meters, RX2 comes back to life and it is fine going to 10 meters.  Looks like you cannot start Thetis with RX2 on 10 meters.

My guess if the band filter selection is wrong in this scenario.

I'm going back to FW 2.1.18 and Thetis 2.10.3.5

**Comment by n1gp** [comment]
@n4py  Seems to working fine here with 2.2.1 & 2.10.3.6, one thing that may be happening
is the ADC for RX2 is not connected to an antenna. Check this setting:

![adcs](https://github.com/ramdor/Thetis/assets/944671/4d134766-62ec-4878-ba8f-fee5b4faea77)

**Comment by n1gp** [comment]
@RdWing 
Here are 2 more images to try:

https://www.dropbox.com/scl/fi/gnzu4s3o9wadb70ikw1ec/Metis_Orion_MkIII_Protocol_2_v2.2.2C.rbf?rlkey=r4ehe4mhuis06uc9rb1rudi9i&dl=0
https://www.dropbox.com/scl/fi/n3npgs5t8es5sjyqhie8s/Metis_Orion_MkIII_Protocol_2_v2.2.2D.rbf?rlkey=irpihahlc21wou9q44h4kstdj&dl=0

**Comment by n4py** [comment]
Neither C nor D will run.  Thetis cannot find the radio with C or D 
firmware.

So I'm back on 2.2.1 with Thetis v2.10.3.6 dev_ant_isolation_test.

With RX2 on if I start thetis, RX2 has no antenna.  The assignments in 
the ADC settings window are correct.  But I can get it to work with the 
BPF window by turning on Bypass for RX2 and then turning off Bypass for RX2.

Carl Moreschi N4PY
127 River Moss Way
Hertford, NC 27944
www.n4py.com

**Comment by n1gp** [comment]
@n4py 

Carl, I believe you own a ANAN-8000, those firmware images I posted are for the MkIII which uses a different FPGA. There are 3 different FPGAs used in the 7000/8000/Andromeda ANAN series. Check this thread for some background info: https://community.apache-labs.com/viewtopic.php?f=27&t=4463

**Comment by n4py** [comment]
Yes I have a 8000dle from 2017. So I'm sure the C and D were wrong for my radio.

It just looks like there's an init step for rx2 missing with the new antenna fix.

73

⁣Carl N4PY

Get BlueMail for Android ​

**Comment by RdWing** [comment]
> @RdWing Here are 2 more images to try:
> 
> https://www.dropbox.com/scl/fi/gnzu4s3o9wadb70ikw1ec/Metis_Orion_MkIII_Protocol_2_v2.2.2C.rbf?rlkey=r4ehe4mhuis06uc9rb1rudi9i&dl=0 https://www.dropbox.com/scl/fi/n3npgs5t8es5sjyqhie8s/Metis_Orion_MkIII_Protocol_2_v2.2.2D.rbf?rlkey=irpihahlc21wou9q44h4kstdj&dl=0

C: Mostly working, sequence errors when operating RX2, transmit spectra still unpure.
D: Constant seq errors, spectra even worse.

**Comment by n4py** [comment]
I think this fix for init on rx2 antenna needs to go into the new thetis.

⁣Carl N4PY

Get BlueMail for Android ​

**Comment by n1gp** [comment]
@RdWing 
I've taken a new approach and made sure the FPGA timing was fully constrained as I was
getting some timing errors. Please try this FW image:
https://www.dropbox.com/scl/fi/6k9yszmrdca1kodj8oqc1/Metis_Orion_MkIII_Protocol_2_v2.2.3A.rbf?rlkey=bfmecsk5chr4aer1u9nrj7fsv&dl=0

@n4py 
Thank-you for pointing out this issue with RX2, your mentioning of
"But I can get it to work with the BPF window by turning on Bypass for RX2 and then turning off Bypass for RX2." got me to relook at my last pull request to ramdor's Thetis source and sure enough, there
is a bug. I put in a new PR to fix it:
https://github.com/ramdor/Thetis/pull/387

**Comment by RdWing** [comment]
> @RdWing I've taken a new approach and made sure the FPGA timing was fully constrained as I was getting some timing errors during the build. Please try this FW image: https://www.dropbox.com/scl/fi/6k9yszmrdca1kodj8oqc1/Metis_Orion_MkIII_Protocol_2_v2.2.3A.rbf?rlkey=bfmecsk5chr4aer1u9nrj7fsv&dl=0
> 
> @n4py Thank-you for pointing out this issue with RX2, your mentioning of "But I can get it to work with the BPF window by turning on Bypass for RX2 and then turning off Bypass for RX2." got me to relook at my last pull request to ramdor's Thetis source and sure enough, there is a bug. I put in a new PR to fix it: #387

Hi Rick,
2.2.3A has Constant SEQ error 8 (DDC2), and transmit spectra still looks funky.

![image](https://github.com/ramdor/Thetis/assets/19214916/106fb2d5-67c9-4edd-a47f-7b9518761983)
![image](https://github.com/ramdor/Thetis/assets/19214916/41409520-96e5-48c0-a8b6-5c7f160bb7e6)

**Comment by ramdor** [comment]
dev_ant_isolation_test_2  here : https://www.dropbox.com/scl/fi/isosv14p0tvguq4a9gzo7/Thetis-v2.10.3.6-dev_ant_isolation_test_2.x64.msi?rlkey=g7z2nnswpkyyava4xjop3d4r6&dl=0

Has fixes from Rick regarding RX2.

VT for the above: https://www.virustotal.com/gui/file/fe24adead0199512bc5526aa164a8e2a056ddafe5b85edb74bf79654ff10039d?nocache=1

**Comment by n4py** [comment]
This version solves the RX2 receive problem and CW transmit on receive 
antenna.

I repeated my test with my LP500 on antenna 3 for receive and a dummy 
load on antenna 1 for transmit.  Transmitted many times on CW.

All is great.  Good job everyone.

Carl Moreschi N4PY
127 River Moss Way
Hertford, NC 27944
www.n4py.com

**Comment by ramdor** [comment]
@n1gp Hey Rick, do you have links to versions of firmware that fix this broken down into radio models and/or fpga variants that we can list in post here. I am not really sure of the current state of everything now tbh. I have been mia for a few weeks and will probably be so until mid/end of Feb. Cheers, Richie.

**Comment by laurencebarker** [comment]
Hi Richie    for Saturn / G2 I can also report that it will be fixed with an upgrade to FPGA V12 and p2app V15. 

 

 

Laurence Barker G8NJJ

***@***.***

**Comment by n1gp** [comment]
@ramdor 

Hi Richie.

Yes, I have firmware images for all of them. Some tested by myself and others. Some
that have not been tested. I posted them to the ANAN Forum here:

https://community.apache-labs.com/viewtopic.php?f=32&t=4972

-Rick / N1GP

**Comment by ramdor** [comment]
@rick @laurencebarker cheers both. I'll tidy up a few things this evening and make another Thetis dev release in a few hrs.

**Comment by rick** [comment]
![jgU8jNQ2ijOxID23He](https://github.com/ramdor/Thetis/assets/6259/5c82eda7-7c90-4d6d-a465-fb47497103c6)

**Comment by Roturbo** [comment]
Hello,

I have ANAN 100D (Angelia) tested  Metis_Angelia_Protocol_2_v12.2.1A.rbf and the first
thing i notice was high SWR, return to previous version and all was fine,, some days later
i make the same test again, and there was the swr problem again.

I´m using Thetis V2.10.3.4, is it necessary to use also latest version ?

Any other process that i must follow ?


Regards
.

**Comment by n1gp** [comment]
@Roturbo 

Please update your Thetis to the latest version: 
https://github.com/ramdor/Thetis/releases/download/v2.10.3.6-dev_2/Thetis-v2.10.3.6-dev_2.x64.msi

There was an issue with the Alex bits not being set correctly in a previous test version.

**Comment by n4py** [comment]
You must also use a special thetis that uses the new antenna selection method in the new firmware. 

⁣Carl N4PY

Get BlueMail for Android ​

**Comment by Roturbo** [comment]
@n1gp 

Hello, i have try latest version, first i test it with old firmware for some hours, all was ok, 
with new firmware i still have erratic SWR, also reset database and it is the same.

If there any other test or firmware to test please inform.


Regards

**Comment by n1gp** [comment]
@Roturbo 

I made 4 more to images to test. I don't have a 100D so I cannot test them myself.
Please let me know what you find:

https://www.dropbox.com/scl/fi/aa98p0qy8bvd8vjnz0s4m/Metis_Angelia_Protocol_2_v12.2.2A.rbf?rlkey=8e23mpbx3vj5l8e9ouhopb4l8&dl=0
https://www.dropbox.com/scl/fi/fcazl1r35xtagdyyh8ot8/Metis_Angelia_Protocol_2_v12.2.2B.rbf?rlkey=35fc1j8kgwkkh4cyj85qqgrth&dl=0
https://www.dropbox.com/scl/fi/sp43s3p4eyjxvuq534l3h/Metis_Angelia_Protocol_2_v12.2.2C.rbf?rlkey=u4c4opvffg698p91bq0d67mom&dl=0
https://www.dropbox.com/scl/fi/3qjff7hl6cyvq0zmajn5z/Metis_Angelia_Protocol_2_v12.2.2D.rbf?rlkey=q3680fo06xeddlp95cu8uhcz9&dl=0

**Comment by n1gp** [comment]
@Roturbo 

Scratch those last four images, they have a bug preventing the Ethernet from working.
Please instead try these four:

https://www.dropbox.com/scl/fi/c7ecqfgk76rnajcks90r8/Metis_Angelia_Protocol_2_v12.2.3A.rbf?rlkey=82agz7bp5jugu4n49tr16op94&dl=0
https://www.dropbox.com/scl/fi/7df896iozqpy6g3gghvin/Metis_Angelia_Protocol_2_v12.2.3B.rbf?rlkey=q7v20otoix2vf1wn3nbzwi758&dl=0
https://www.dropbox.com/scl/fi/dzv889isv2q017s8aguaa/Metis_Angelia_Protocol_2_v12.2.3C.rbf?rlkey=2rjqjygsnq5g7tjwdglf590dt&dl=0
https://www.dropbox.com/scl/fi/tbzxc2wm9cyj42kh741sk/Metis_Angelia_Protocol_2_v12.2.3D.rbf?rlkey=ychcj6rw1pfhzc989cq74vhc2&dl=0

**Comment by Roturbo** [comment]
Hello @n1gp 

**Metis_Angelia_Protocol_2_v12.2.3J.rbf**
Less SWR spikes but still pressent
Sequence error  266 (CCDDC2 Mic) with audio interrupts


**Metis_Angelia_Protocol_2_v12.2.3I.rbf**
Too many sequence error, didn´t test the SWR
Sequence error   264  (DDC2 Mic)  connection lost, Thetis shutdown 


**Metis_Angelia_Protocol_2_v12.2.3K.rbf**
No connection.

**Metis_Angelia_Protocol_2_v12.2.3L.rbf**
No connection.


Back to **Metis_Angelia_Protocol_2_v12.1.8.rbf** and all starts to work again.

Regards

**Comment by n1gp** [comment]
Thanks @Roturbo 

I have gone back to the 12.1.8 firmware and added in only the CW/RF fix. Here are
4 more images from that:

https://www.dropbox.com/scl/fi/qseteidyov0mvhe82uzv8/Metis_Angelia_Protocol_2_v12.2.4A.rbf?rlkey=xdt3x620t9h4p4kcyxru2we7t&dl=0
https://www.dropbox.com/scl/fi/xnkmvrpbyd01qlgcjgs9i/Metis_Angelia_Protocol_2_v12.2.4B.rbf?rlkey=o7ck17ox4de0y4kb5qmucserd&dl=0
https://www.dropbox.com/scl/fi/jb0efjcrrpcuhwe8shk9a/Metis_Angelia_Protocol_2_v12.2.4C.rbf?rlkey=2ysrva8piaolt13k50ggerf0u&dl=0
https://www.dropbox.com/scl/fi/4rpwl3ipkuc455vhfn6j3/Metis_Angelia_Protocol_2_v12.2.4D.rbf?rlkey=afv4omz5qonqxkvjm15chf4rm&dl=0

**Comment by Roturbo** [comment]
.
Hi @n1gp 

here is the report.

**Metis_Angelia_Protocol_2_v12.2.4E.rbf**
SWR ok on TUN, but didn´t test with 2TON ,, no PS, Single Cal nor responding


**Metis_Angelia_Protocol_2_v12.2.4F.rbf**
SWR 1.7 with TUN,, with 2TON SWR is more than 2 and RED SQUARE is back,, 
PS not working, only some random dots, Single Cal nor responding



**Metis_Angelia_Protocol_2_v12.2.4G.rbf**
SWR ok on TUN and 2TON,, PS start to work something but blue and yellow 
lines are dots instead of lines

**Metis_Angelia_Protocol_2_v12.2.4H.rbf**
No SWR on TUN or 2TON, PS working normal, AmpView lines perfect.


The last one (H) seams to solve the 2 problems i have found, i will keep using this version
and if there is any other problem i will report, i don´t do CW so the problem of power on
external RX ports was not tested.

Tanks for your work.

Regards

**Comment by Roturbo** [comment]
Hi @n1gp 

found one problem, PS after some time get confused and the main signal is ok, but it start
to project interference 200khz up and down from the center audio  (SSB).

They are mirror, for example if i see a extra signal on left side at 40khz i do see 
the same type of interference 40khz on the right.

At start all is ok,, after some minutes of talking and it start to do this, 
maybe timing or temperature related, but i was using less than 20w.

Also, on the moment i see the PS disturb for the second time with the space of 3 hours
i run to change back to the v12.1.8 without changing any setting, and with the difference
of only 2 or 3 minutes and with the new firmware all start to work normal.

I keep talking and the problem was solved,, so the v12.2.4H is the best from all,
still it need some tuning on PS,, or maybe in other points that i still didn´t see.

Regards

**Comment by Roturbo** [comment]
![image](https://github.com/ramdor/Thetis/assets/91867260/8f840073-a380-4dcd-b8d2-9903a188903e)
Sample Rate is 768000

**Comment by ramdor** [comment]
Since there have been no updates since 20Feb, I will close this. We can re-open if required.
